### ISSUE TITLE
Bootstrap - support/switch to higher version and themes

### ISSUE THREAD

**Issue by Polkas** [insightsengineering/teal#617]
TL;DR

-  shiny::fluidPage(theme = bslib::bs_theme(version = "4"), ...), a little updated shiny app view. Mostly it is adding more tools to write in HTML/CSS/JS for the developer of the app. Bootstrap 4 breaks current teal.
- shiny::fluidPage(theme = bslib::bs_theme(bg = "black", fg = "white", primary = "purple"), ...). Specify by the user/app developer. It requires bootstrap 4, so breaks current apps.

![image](https://user-images.githubusercontent.com/10676545/162713652-b89df445-cd67-42e1-a009-d7a5dbd1443d.png)

The `shiny` package is using the bootstrap version 3 and most probably it will be used for backward compatibility.
Supporting many bootstrap versions in NEST at the same time might be more complex as we have some js/bootstrap code.
We could focus to use certain new version 4, the version 5 is still in development so we should be carefull with it.
**Why to bother about new bootstrap versions.**
We gain more functionalities which are global standard for the UI.
What was updated in the version 4 compared to 3:
https://getbootstrap.com/docs/4.6/migration/

How the new bootstrap version could be set in the shiny app, https://shiny.rstudio.com/reference/shiny/1.7.0/bootstrapPage.html.
In the case of  the `shiny::fluidPage` function we are adding additional `theme` argument where we could add a proper bootstrap dependency. RStudio is recommending usage of `bslib::bs_theme` function, like `bslib::bs_theme(version = "4")`.

In the `teal` case we are using `fluidPage` here:
https://github.com/insightsengineering/teal/blob/15b7d4c06ae51a83d801b8badc86bff286f2608e/R/module_teal.R#L95

The `bslib`  package offers many themes which could be added with the one line of code to any shiny app and improve the user experience. https://rstudio.github.io/bslib/articles/bslib.html#custom

Please try:

```
library(bslib)
bs_theme_preview()
```

**How much bootstrap 4 influence current apps:**

To check how much the bootstrap 4 breaks our apps please update the `teal` package with
`theme = bslib::bs_theme(
        # Controls the default grayscale palette
        bg = "#202123", fg = "#B8BCC2",
        # Controls the accent (e.g., hyperlink, button, etc) colors
        primary = "#EA80FC", secondary = "#48DAC6",
        base_font = c("Grandstander", "sans-serif"),
        code_font = c("Courier", "monospace"),
        heading_font = "'Helvetica Neue', Helvetica, sans-serif",
        # Can also add lower-level customization
        "input-border-color" = "#EA80FC"
      ),` code, added here https://github.com/insightsengineering/teal/blob/15b7d4c06ae51a83d801b8badc86bff286f2608e/R/module_teal.R#L95

and run the Exploratory App.

**Comment by donyunardi** [comment]
I will treat this as Roadmap card and remove NESTDevs project association from this issue.

- https://github.com/insightsengineering/teal/issues/709
- https://github.com/insightsengineering/teal/issues/719

**Comment by nikolas-burkoff** [comment]
Can be closed by the PRs here https://github.com/insightsengineering/teal/issues/719#issuecomment-1249488162
